### DISCUSSION TITLE
Lifetime of import statements and variables

### DISCUSSION THREAD

**Discussion by ssimo3lsuhsc** [tndrle/PyLuaTeX#21] [Q&A]
I have a long document that will potentially call features of PyLuaTeX multiple times. I want to clarify something. I do not have any reason to change PyLuaTeX's `shutdown` option. I'm keeping it as `veryveryend`. I want to be clear of the secondary implications of keeping it the way it is.

Particularly, I want to be clear if I am correct in assuming that if I issue a statement `foo = None` or `import bar` in one block of Python code, then I do _not_ have to issue the same statements in subsequent blocks of code.

**Comment by tndrle** [comment]
Your assumption is correct. Python state is retained until the end of the document.

The effect of changing the `shutdown` option is very subtle and not noticeable in most scenarios.
PyLuaTeX (or, more precisely, the connected Python instance) shuts down ...
* `veryenddocument`: ... after the `.aux` file is closed and read in again
* `veryveryend`: ... right before the final `\@@end`
* `off`: ... after the LuaTeX process shuts down

Early versions of PyLuaTeX behaved like `off` but that caused problems with SyncTeX under Windows (#8).
